PhotoAlbumSample/app/src/main/java/com/amazonaws/android/samples/photosharing/AppSyncHelper.java

```diff
@@ -0,0 +1,225 @@
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.android.samples.photosharing;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.amazonaws.amplify.generated.graphql.CreateAlbumMutation;
+import com.amazonaws.amplify.generated.graphql.CreatePhotoMutation;
+import com.amazonaws.amplify.generated.graphql.DeleteAlbumMutation;
+import com.amazonaws.amplify.generated.graphql.ListAlbumsQuery;
+import com.amazonaws.amplify.generated.graphql.UpdateAlbumMutation;
+import com.amazonaws.mobile.client.AWSMobileClient;
+import com.amazonaws.mobile.config.AWSConfiguration;
+import com.amazonaws.mobileconnectors.appsync.AWSAppSyncClient;
+import com.amazonaws.mobileconnectors.appsync.fetcher.AppSyncResponseFetchers;
+import com.amazonaws.mobileconnectors.appsync.sigv4.CognitoUserPoolsAuthProvider;
+import com.apollographql.apollo.GraphQLCall;
+import com.apollographql.apollo.api.Response;
+import com.apollographql.apollo.exception.ApolloException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+import javax.annotation.Nonnull;
+
+import type.CreateAlbumInput;
+import type.CreatePhotoInput;
+import type.DeleteAlbumInput;
+import type.UpdateAlbumInput;
+
+/**
+ * AppSyncHelper maintains utilities for using AppSync.
+ */
+public class AppSyncHelper {
+
+    private static final String TAG = AppSyncHelper.class.getSimpleName();
+
+    private AWSAppSyncClient mAWSAppSyncClient;
+
+    /**
+     * Initialize an AWSAppSyncClient when create an AppSyncHelper object.
+     * @param context
+     */
+    public AppSyncHelper(Context context) {
+
+        // Initialize AWSAppSyncClient
+        mAWSAppSyncClient = AWSAppSyncClient.builder()
+                .context(context)
+                .awsConfiguration(new AWSConfiguration(context))
+                .cognitoUserPoolsAuthProvider(new CognitoUserPoolsAuthProvider() {
+                    @Override
+                    public String getLatestAuthToken() {
+                        try {
+                            return AWSMobileClient.getInstance().getTokens().getIdToken().getTokenString();
+                        } catch (Exception e) {
+                            Log.e(TAG, e.getLocalizedMessage());
+                            return null;
+                        }
+                    }
+                })
+            .build();
+
+    }
+
+    private GraphQLCall.Callback<CreateAlbumMutation.Data> createAlbumCallback = new GraphQLCall.Callback<CreateAlbumMutation.Data>() {
+        @Override
+        public void onResponse(@Nonnull Response<CreateAlbumMutation.Data> response) {
+            String albumName = response.data().createAlbum().name();
+            Log.i(TAG, "An album with name" + albumName + "is created successfully.");
+        }
+
+        @Override
+        public void onFailure(@Nonnull ApolloException e) {
+            Log.i(TAG, "Failed to create an album.");
```

Comment:
Add an AlertDialog when there is a failure to report to the user of the sample app. Repeat this for all success and failure scenarios.